great_expectations/rule_based_profiler/domain_builder/column_domain_builder.py

```diff
@@ -1,23 +1,54 @@
-from typing import List, Optional
+from typing import List, Optional, Union
 
+import great_expectations.exceptions as ge_exceptions
+from great_expectations.core.batch import BatchRequest, RuntimeBatchRequest
 from great_expectations.execution_engine.execution_engine import MetricDomainTypes
 from great_expectations.rule_based_profiler.domain_builder import DomainBuilder
 from great_expectations.rule_based_profiler.types import Domain, ParameterContainer
 from great_expectations.validator.metric_configuration import MetricConfiguration
 
 
 class ColumnDomainBuilder(DomainBuilder):
+    def __init__(
+        self,
+        data_context: "DataContext",  # noqa: F821
+        batch_request: Optional[Union[BatchRequest, RuntimeBatchRequest, dict]] = None,
+        column_names: Optional[List[str]] = None,
+    ):
+        """
+        Args:
+            data_context: DataContext
+            batch_request: specified in DomainBuilder configuration to get Batch objects for domain computation.
+        """
+
+        super().__init__(
+            data_context=data_context,
+            batch_request=batch_request,
+        )
+
+        self._column_names = column_names
+
+    @property
+    def domain_type(self) -> Union[str, MetricDomainTypes]:
+        return MetricDomainTypes.COLUMN
```

Comment:
Isn't this just a `MetricDomainType`?

Reply:
@cdkini Yes -- and it is defined as the `MetricDomainTypes` enum in `ExecutionEngine`.

Reply:
So I don't think we should annotate with `str`

Reply:
@cdkini I got PyCharm errors... not sure why.